persistence/jpa/src/main/java/org/infinispan/persistence/jpa/JpaStore.java

```diff
@@ -423,14 +423,34 @@ public void write(MarshallableEntry entry) {
    }
 
    @Override
-   public void writeBatch(Iterable<MarshalledEntry<? extends K, ? extends V>> marshalledEntries) {
-      EntityManager em = emf.createEntityManager();
-      try {
-         EntityTransaction txn = em.getTransaction();
-         long txnBegin = timeService.time();
-         try {
-            txn.begin();
-            for (MarshallableEntry entry : marshalledEntries) {
+   public void writeBatch(Publisher<MarshallableEntry<? extends K, ? extends V>> publisher) {
+      long txnBegin = timeService.time();
```

Comment:
Technically this should be initialized in the first using block. A Publisher is supposed to be able to be subscribed to multiple times. This would mean if it was retained and reused later it would have the wrong timestamp.

Reply:
If we do that I'm not sure how we would be able to pass the start time to the error handler for `stats.addBatchWriteTxFailed`

Reply:
You can pass it by changing the inner publisher's returned value to a single value and mapping that to the time.
`
.takeLast(1).map(ignore -> startTime)
`
I am sure there is a better way to do this, but can't think of it atm. Maybe encapsulating the whole Publisher in a flatMap operation or something.